crates/syntax/src/ast/token_ext.rs

```diff
@@ -571,29 +566,30 @@ impl ast::IntNumber {
 
     pub fn suffix(&self) -> Option<&str> {
         let text = self.text();
-        // FIXME: don't check a fixed set of suffixes, `1_0_1_l_o_l` is valid
-        // syntax, suffix is `l_o_l`.
-        ast::IntNumber::SUFFIXES.iter().chain(ast::FloatNumber::SUFFIXES.iter()).find_map(
-            |suffix| {
-                if text.ends_with(suffix) {
-                    return Some(&text[text.len() - suffix.len()..]);
-                }
-                None
-            },
-        )
+        let radix = self.radix();
+        let mut indices = text.char_indices();
+        if radix != Radix::Decimal {
+            indices.next()?;
+            indices.next()?;
+        }
```

Comment:
This could be rewritten
```suggestion
        let mut indices = text.char_indices().skip(if radix == Radix::Decimal { 0 } else { 2 });
```
but I'm not sure what I prefer.

Reply:
the current versions LGTM.

Let's add a couple of unit tests though, I *think* this handles `_` correctly, but I'd trust a unit test more than myself :D

Reply:
Okay, I think these tests should cover everything.